BMM.Core/Implementations/Analytics/AppCenterAnalytics.cs

```diff
@@ -24,8 +28,19 @@ public void LogEvent(string eventName)
         public void LogEvent(string eventName, IDictionary<string, object> parameters)
         {
             var user = _userStorage.GetUser();
-            if (!parameters.ContainsKey("PersonId") && user != null)
-                parameters.Add("PersonId", _userStorage.GetUser().PersonId);
+            if (_remoteConfig.UseAnalyticsId)
+            {
+                if (!parameters.ContainsKey(nameof(User.AnalyticsId)) && user != null)
+                    parameters.Add(nameof(User.AnalyticsId), user.AnalyticsId);
+            }
+            else
+            {
+                if (!parameters.ContainsKey(nameof(user.PersonId)) && user != null)
+                    parameters.Add(nameof(user.PersonId), user.PersonId);
+            }
+
+            if (user?.Age != null)
+                parameters.Add(nameof(user.Age), user.Age);
```

Comment:
I would do it that way, to not copy the same check:

```
        public void LogEvent(string eventName, IDictionary<string, object> parameters)
        {
            var user = _userStorage.GetUser();

            void AddParameter(string key, object value)
            {
                if (!parameters.ContainsKey(key) && user != null)
                    parameters.Add(key, value);
            }

            if (_remoteConfig.UseAnalyticsId)
                AddParameter(nameof(User.AnalyticsId), user.AnalyticsId);
            else
                AddParameter(nameof(user.PersonId), user.PersonId);

            (...)
        }

```

Reply:
nice. refactored it